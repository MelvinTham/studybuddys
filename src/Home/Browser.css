.browserTitle{
    height: 100vh;
    margin: 0;
    /* display: flex;
    flex-direction: column;     
    justify-content: center; 
    align-items: center;   */
    background-image: linear-gradient(to right top, #e00a0a, #e44900, #e66b00, #e58800, #e4a300);
    font-family:Nunito,'Times New Roman', Times, serif;
  }
  .cardSize{
    width:600px;
    height:auto;
    margin-top: 20px;
  }
  .newBrowserTitle{
    height: 100vh;
    margin: 0;
    display: flex;
    flex-direction: column;     
    justify-content: center; 
    align-items: center;  
    background-image: linear-gradient(to right top, #e00a0a, #e44900, #e66b00, #e58800, #e4a300);
    font-family:Nunito,'Times New Roman', Times, serif;
  }