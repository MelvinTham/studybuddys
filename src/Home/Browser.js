import React, {Component} from 'react';
import Button from '@material-ui/core/Button';
import MenuItem from '@material-ui/core/MenuItem';
import Typography from '@material-ui/core/Typography';
import GridList from '@material-ui/core/GridList';
import TextField from '@material-ui/core/TextField';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Navbar from "./Navbar";
import axios from 'axios';
import * as firebase from 'firebase';
import './Dashboard.css';
import './Browser.css';

class Browser extends Component {
    constructor(props) {
        super(props);
        this.state = {
          school:'',
          subject:'',
          class:[],
          userClasses: [],
          loading: true,
        }
        this.handleChange = this.handleChange.bind(this);
        this.addCourseToUser = this.addCourseToUser.bind(this);
        this.getCourseName = this.getCourseName.bind(this);
        this.checkLoggedIn = this.checkLoggedIn.bind(this);
    }
    state = {
        anchorEl: null,
      };

      // handleClick = event => {
      //   this.setState({ anchorEl: event.currentTarget });
      // };

      handleClick(event) {
        this.setState({userClasses: event.target.value})
      }

      handleClose = () => {
        this.setState({ anchorEl: null });
      };

      handleChange = name => event => {
        this.setState({
          [name]: event.target.value,
        });
      };

      componentDidMount(){
        this.checkLoggedIn();
      }

      checkLoggedIn(){
        var prop = this.props;
        firebase.auth().onAuthStateChanged(function(user) {
           if (!user) {
              prop.history.push('/');
           }
        });
     }

      getCourseName(e){
          console.log("Clicked");
          var info = { // JSON object to pass to the api call
              schoolName: this.state.school,
              subject: this.state.subject
          };
          axios.post('https://triple-bonito-221722.appspot.com/api/getCourses', info)
          .then(response => {
              console.log("Response: " + response.data.courseID);
              this.setState({class : response.data});
          })
          this.setState({loading: false});
      }

      addCourseToUser(e){
         e.preventDefault();
         var course = e.currentTarget.value;
         firebase.auth().onAuthStateChanged(function(user) {
            if (user) {
               var info = {
                  id: user.uid,
                  courseName: course
               };
               axios.post('https://triple-bonito-221722.appspot.com/api/addCourses', info)
            }
         });
      }

      render() {

        console.log(this.state.class)
        const schools = [
            {
                value: '',
                label: '',
            },
            {
              value: 'CUNY CCNY',
              label: 'The City College Of New York',
            },
            {
              value: 'CUNY Baruch',
              label: 'Baruch College',
            },
            {
              value: 'CUNY York',
              label: 'York College',
            },
            {
              value: 'CUNY Queens',
              label: 'Queens College',
            },
          ];

          // const subject = [
          //   {
          //       value: '',
          //       label: '',
          //   },
          //   {
          //     value: 'Art',
          //     label: 'Art',
          //   },
          //   {
          //     value: 'Bio',
          //     label: 'Bio',
          //   },
          //   {
          //     value: 'Csc',
          //     label: 'Csc',
          //   },
          //   {
          //     value: 'Eco',
          //     label: 'Eco',
          //   },
          // ];

        console.log(this.state.school);
        console.log(this.state.subject);
        console.log(this.state.class);
        console.log(this.state.userClasses);

          let loading = (null)
          if (this.state.loading === false) {
            loading = (<h1>Loading...</h1>)
          }

        return (

            <div className="browserTitle">
            <div className="nav">
             <Navbar />
            </div>
                <div className = "flexCenter">
                    <h1
                      style = {{color: "black", marginTop: "100px",fontSize:'60px',height:'15vh'}}
                      data-aos="fade-down"
                      data-aos-easing="linear"
                      data-aos-duration="400"> Select a Class
                    </h1>


                <div style = {{height:'65vh'}}>
                <TextField
                    select
                    label="Select"
                    value={this.state.school}
                    onChange={this.handleChange('school')}
                    helperText="Please select your school"
                    margin="normal"
                    variant="outlined"
                    data-aos="fade-down"
                    data-aos-easing="linear"
                    data-aos-duration="400">

                    {schools.map(option => (
                      <MenuItem key={option.value} value={option.value}>
                          {option.label}
                      </MenuItem>
                    ))}
                </TextField>

                <br/>
                <div className="flexCenter">
                 {
                    this.state.school !== ''?
                      <TextField
                        required
                        label="Select"
                        value={this.state.subject}
                        onChange={this.handleChange('subject')}
                        helperText="Please select your Subject"
                        margin="normal"
                        variant="outlined"
                        data-aos="fade-left"
                        data-aos-easing="linear"
                        data-aos-duration="400"
                      >
                      </TextField>

                : null

                 }
                 </div>
                 {this.state.school !==''?
                    <Button
                        onClick={(this.getCourseName)}
                        type = 'submit'
                        >
                        Submit
                    </Button>:null

                 }
                 <br/>

                 {
                  this.state.class.courseID === undefined?
                     loading :
                    <div className="flexRow" data-aos="fade-down" data-aos-easing="linear" data-aos-duration="500">
                       <GridList  cols={3} padding={150} >
                       {console.log("Course list: " + this.state.class.courseID)}
                         {this.state.class.courseID.map((data, key) => {
                          return(

                              <Card key = {key} value={data} className ="flexRow" style={{width:'250px',height:'250px',margin:'10px 10px'}}>
                                  <form key = {key}>
                                    <CardContent>
                                        <Typography variant ="headline">
                                          {data}
                                        </Typography >
                                        <Typography variant ="headline">

                                          <Button
                                            onClick={(e)=>this.addCourseToUser(e)}
                                            value={data}
                                            type = "submit"
                                            variant = "outlined"
                                            style = {{marginTop:'15px'}}
                                          >
                                            Add Course
                                          </Button>

                                        </Typography >
                                    </CardContent>
                                  </form>
                              </Card>
                            )
                          })}

                      </GridList>
                    </div>

                  }
                  </div>
              </div>

            </div>
        )
    }
}

export default Browser;
