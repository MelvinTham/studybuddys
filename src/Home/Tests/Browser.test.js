import Enzyme, { render, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import React from 'react';
import Browser from './../Browser';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });
const getDefaultProps = () => ({});
describe('Browser component', () => {
  it('shallow renders without crashing', () => {
    const {} = getDefaultProps();
    shallow(
        <Router>
            <Browser />
        </Router>);
  });
  it('render snapshot', () => {
    const {} = getDefaultProps();
    const wrapper = render(
        <Router>
            <Browser />
        </Router>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
