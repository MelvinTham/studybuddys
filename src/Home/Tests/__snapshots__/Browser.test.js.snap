// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browser component render snapshot 1`] = `
<div
  class="browserTitle"
>
  <div
    class="nav"
  >
    <div>
      <button
        class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7"
        style="float:left"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-3"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-31 MuiSvgIcon-fontSizeLarge-39"
            focusable="false"
            role="presentation"
            style="color:white"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      class="flexRow"
      style="margin-left:auto"
    >
      <span
        style="font-size:22px;margin-right:10px"
      >
        <a
          href="/dashboard/profile"
          style="color:white"
        />
      </span>
      <button
        class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7"
        style="color:white;font-size:17px"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-3"
        >
          Logout
        </span>
      </button>
    </div>
  </div>
  <div
    class="flexCenter"
  >
    <h1
      data-aos="fade-down"
      data-aos-duration="400"
      data-aos-easing="linear"
      style="color:black;margin-top:100px;font-size:60px;height:15vh"
    >
      Select a Class
    </h1>
    <div
      style="height:65vh"
    >
      <label
        class="MuiFormLabel-root-65 MuiInputLabel-root-54 MuiInputLabel-animated-62"
      >
        Select School
        <div
          class="MuiInputBase-root-92 MuiInput-root-79 MuiInput-underline-83"
          style="width:300px"
        >
          <div
            class="MuiSelect-root-72"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              class="MuiSelect-select-73 MuiSelect-selectMenu-76 MuiInputBase-input-102 MuiInput-input-87"
              role="button"
              tabindex="0"
            >
              <span>
                â€‹
              </span>
            </div>
            <input
              id="SelectSchool"
              name="school"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-31 MuiSelect-icon-78"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </label>
      <br />
      <div
        class="flexCenter"
      />
      <br />
    </div>
  </div>
</div>
`;
